src/main/java/org/apache/commons/lang3/math/NumberUtils.java

```diff
@@ -1374,8 +1407,11 @@ public static boolean isNumber(final String str) {
         boolean hasDecPoint = false;
         boolean allowSigns = false;
         boolean foundDigit = false;
+        boolean isJava6 = StringUtils.startsWith(
+                System.getProperty("java.version"), "1.6");
```

Comment:
You can better use `SystemUtils`for this. But I'm about to extend `JavaVersion` in for [LANG-1263](https://issues.apache.org/jira/browse/LANG-1263) so that will make this even easier. I'll ping you when I have that finished.